### ISSUE TITLE
Github workflow failure

### ISSUE THREAD

**Issue by dihm** [labscript-suite/labscript-suite#69]
All github workflows have been broken on the Build (ubuntu-latest, 3.8, x64) step. See [here](https://github.com/labscript-suite/labscript-devices/actions/runs/2408755764) for a representative example (with a second run that has debug logging enabled).

Doing some local testing, it appears to be the result of the build script installing `setuptools-conda` from the `cbillington` conda channel (which isn't maintained in favor of the `labscript-suite` channel). So the fix appears to be to change which channel we install from in all the repo and workflows.

On a related note, the workflow also manually installs the updated version of `setuptools-scm` from pip. I believe this was necessary to fix a bug using version 4.1.0 which wasn't available on conda at the time (since conda packages update pretty slowly). It does not appear to be necessary anymore, so I plan to remove this step as well.

Is there any other random workflow things we should address while I muck about? I could see an argument for moving the artifact upload to github before the testPyPI upload to prevent pip version lock on failed runs.

Checklist to address:

- [x] labscript-suite
- [x] labscript
- [x] labscript-devices
- [x] labscript-utils
- [x] blacs
- [x] lyse
- [x] runmanager
- [x] runviewer

**Comment by chrisjbillington** [comment]
The state of the art for what these workflows are supposed to be is workflow-sandbox's release workflow:

https://github.com/rpanderson/workflow-sandbox/blob/master/.github/workflows/release.yml

Which I have kept up to date with the changes needed to fix the various issues I've come up against in individual repositories. Clearly I failed to roll it out to all the repos that needed it.

I like to diff workflow-sandbox's workflow against individual projects to check if they're up-to-date with the latest, for most projects the only differences should be references to the project name/repository, anaconda username, and which options are selected for the `PURE` and `NOARCH` variables.

Actually have a half-finished bit of code to roll out the latest workflow as a template from workflow-sandbox to other projects. Automating the automation...yuck. But better than projects' workflows bit rotting.
